### PR TITLE
Sabrina/re-structuring project

### PR DESCRIPTION
**Pull request by sabrina-li**
Re-structuring the project by creating a project and then integrate pod, this shared the scheme in the project rather than in pod project, so we can support both cocoapods and carthage
Let me know if this makes more sense.

### CODE REVIEW

File: FullStorySegmentMiddleware.podspec [hunk 1]
```diff
@@ -1,38 +1,133 @@
...
+  #  can feel like a chore to fill in it's definitely to your advantage. The
+  #  summary should be tweet-length, and the description more in depth.
+  #
 
-  s.description      = <<-DESC
-TODO: Add long description of the pod here.
-                       DESC
+  spec.name         = "FullStorySegmentMiddleware"
+  spec.version      = "0.1.0"
+  spec.summary      = "FullStory Segment Middleware is a project demonstrating the full potential of joining forces between Segment + FullStory."
```
Comment (nvzqz) [inline FullStorySegmentMiddleware.podspec]: I think the summary should be a bit shorter.
```suggestion
  spec.summary      = "Demonstrates the full potential of joining forces between Segment and FullStory."
```

File: FullStorySegmentMiddleware.podspec [hunk 2]
```diff
@@ -1,38 +1,133 @@
...
+  # This description is used to generate tags and improve search results.
+  #   * Think: What does it do? Why did you write it? What is the focus?
+  #   * Try to keep it short, snappy and to the point.
+  #   * Write the description between the DESC delimiters below.
+  #   * Finally, don't worry about the indent, CocoaPods strips it!
+  spec.description  = <<-DESC
+  FullStory Segment Middleware is a project demonstrating the full potential of joining forces between Segment + FullStory.
```
Comment (nvzqz) [inline FullStorySegmentMiddleware.podspec]: Likewise I think the description should be shorter.

```suggestion
  This demonstrates the full potential of joining forces between Segment and FullStory.
```

File: FullStorySegmentMiddleware.podspec [hunk 3]
```diff
@@ -1,38 +1,133 @@
...
+  TODO: detailed description needed here
+                   DESC
 
-  s.homepage         = 'https://github.com/fullstorydev/fullstory-segment-middleware-ios'
-  # s.screenshots     = 'www.example.com/screenshots_1', 'www.example.com/screenshots_2'
-  s.license          = { :type => 'MIT', :file => 'LICENSE' }
-  s.author           = { 'FullStory' => 'https://www.fullstory.com' }
-  s.source           = { :git => 'https://github.com/fullstorydev/fullstory-segment-middleware-ios.git', :tag => s.version.to_s }
+  spec.homepage     = "https://github.com/fullstorydev/fullstory-segment-middleware-ios"
+  # spec.screenshots  = "www.example.com/screenshots_1.gif", "www.example.com/screenshots_2.gif"
```
Comment (nvzqz) [inline FullStorySegmentMiddleware.podspec]: I would just remove this line.

File: FullStorySegmentMiddleware.podspec [hunk 4]
```diff
@@ -1,38 +1,133 @@
...
+
+
+  # ――― Platform Specifics ――――――――――――――――――――――――――――――――――――――――――――――――――――――― #
+  #
+  #  If this Pod runs only on iOS or OS X, then specify the platform and
+  #  the deployment target. You can optionally include the target after the platform.
+  #
+
+  # spec.platform     = :ios
+  spec.platform     = :ios, "8.0"
```
Comment (nvzqz) [inline FullStorySegmentMiddleware.podspec]:
```suggestion
  spec.platform     = :ios, "8.0"
```

File: FullStoryMiddleware/FullStoryMiddleware.h [hunk 5]
```diff
@@ -12,6 +12,13 @@
...
+//! Project version number for FullStoryMiddleware.
+FOUNDATION_EXPORT double FullStoryMiddlewareVersionNumber;
+
+//! Project version string for FullStoryMiddleware.
+FOUNDATION_EXPORT const unsigned char FullStoryMiddlewareVersionString[];
+
+
 @interface FullStoryMiddleware : NSObject<SEGMiddleware>
 
 @property (strong, nonatomic) NSArray* allowlistEvents;
```
Comment (nvzqz) [inline FullStoryMiddleware/FullStoryMiddleware.h]: While we're restructuring the project...
```suggestion
@property (strong, nonatomic) NSArray<NSString *>* allowlistEvents;
```

File: FullStoryMiddlewareExample/main.m [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+#import <UIKit/UIKit.h>
+#import "AppDelegate.h"
+
+int main(int argc, char * argv[]) {
+    NSString * appDelegateClassName;
+    @autoreleasepool {
+        // Setup code that might create autoreleased objects goes here.
+        appDelegateClassName = NSStringFromClass([AppDelegate class]);
+    }
```
Comment (nvzqz) [inline FullStoryMiddlewareExample/main.m]: This autorelease pool isn't necessary.

File: Podfile [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+
+target 'FullStoryMiddlewareTests' do
+  common_pods
+  pod 'OCMock', '~>3.6'
+end
+
+target 'FullStoryMiddleware' do
+  common_pods
+end
```
Comment (nvzqz) [inline Podfile]: I would sort these in order of relevance.

```suggestion
target 'FullStoryMiddleware' do
  common_pods
end

target 'FullStoryMiddlewareTests' do
  common_pods
  pod 'OCMock', '~>3.6'
end

target 'FullStoryMiddlewareExample' do
  common_pods
  pod 'FullStorySegmentMiddleware', :path => '.'
end
```

**Review by nvzqz — CHANGES_REQUESTED** [review]
I think this makes the project easier to work with and inspect as an outsider. Good job!

I left some comments/suggestions on things I think could be improved.

File: FullStoryMiddlewareExample/main.m (hunk 6 above)
Comment (sabrina-li) [reply FullStoryMiddlewareExample/main.m]: removed

File: FullStorySegmentMiddleware.podspec (hunk 3 above)
Comment (sabrina-li) [reply FullStorySegmentMiddleware.podspec]: removed

**Comment by sabrina-li** [comment]
> I think this makes the project easier to work with and inspect as an outsider. Good job!
> 
> I left some comments/suggestions on things I think could be improved.

Thanks so much for the detailed review!! updated. please give this another go :)

File: FullStoryMiddlewareExample/main.m [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by FullStory on 9/24/20.
+//  Copyright (c) 2020 FullStory. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import "AppDelegate.h"
+
+int main(int argc, char * argv[]) {
+    NSString * appDelegateClassName;
```
Comment (nvzqz) [inline FullStoryMiddlewareExample/main.m]: You removed the autorelease pool, but you also removed getting the name of the app delegate's class.
```suggestion
    NSString * appDelegateClassName = NSStringFromClass([AppDelegate class]);
```

**Review by nvzqz — CHANGES_REQUESTED** [review]
Thanks for making the requested changes! Although you accidentally removed some important code while doing so.

**Review by nvzqz — APPROVED** [review]
🚢 it

**Pull request merged**